outrank/core_ranking.py

```diff
@@ -59,6 +60,36 @@ def prior_combinations_sample(combinations: list[tuple[Any, ...]], args: Any) ->
     return tmp
 
 
+def get_combinations_from_columns(all_columns: pd.Index, args: Any) -> list[tuple[Any, ...]]:
+    """Return feature-feature & feature-label combinations, depending on the heuristic and ranking scope"""
+
+    if '3mr' in args.heuristic:
+        rel_columns = [column for column in all_columns if ' AND_REL ' in column]
+        non_rel_columns = sorted(set(all_columns) - set(rel_columns))
+
+        combinations = list(
+            itertools.combinations_with_replacement(non_rel_columns, 2),
+        )
+        combinations += [(column, args.label_column) for column in rel_columns]
+    else:
+        _combinations = itertools.combinations_with_replacement(all_columns, 2)
```

Comment:
You could also just overwrite `combinations` in case target_ranking_only, this would remove the else part alltogether (and unburden the gc a bit I guess)